src/src-assets/creator-assets/services.coffee

```diff
@@ -404,6 +462,75 @@ Adventure.service "treeSrv", ['$rootScope','$filter','$sanitize','legacyQsetSrv'
 
 		return
 
+	updateAllItems = (tree, updatedItem) ->
```

Comment:
This one is confusing me a little bit - I assumed a given node would only need a reference to an item ID, and a "master list" of items would contain each item's actual properties: the name, description, and icon. Are those properties also being stored per-node wherever an item is associated with a node? That seems overly complicated - ideally, the item ID remains static (sort of like node IDs), and only the properties of the item in the "master list" would need updating. Feel free to clarify if I'm wrong here.